### PR TITLE
Introduce optin & minimalistic IO mocks

### PR DESCRIPTION
**Pull request by Annih**
In some cases your code or one of its dependencies uses IO methods instead of File ones - which is OK as File inherits from IO.

As IO class is kind of low level, it can be tricky to mock its methods.
So to ensure it does not break all FakeFS users, let's use an opt-in mechanism!

If you use FakeFS.activate!(io_mocks: true) you'll get some IO methods mocked!

### CODE REVIEW

**Comment by Annih** [comment]
Inspired by #395 and could solve #461

**Comment by Annih** [comment]
Feel free to advise on README update or tests to add.

**Comment by grosser** [comment]
opt-in sounds great to get this some real world usage ... some basic tests + readme instructions would be nice :)

**Comment by Annih** [comment]
I'll try to work on tests this week; it should be rather easy as I was already using a similar mechanism in my tests.

As the author of #395 my case was kind of Chef related, I wonder if other ruby apps are using IO 🤔

**Comment by Annih** [comment]
OK let me know whether the tests are enough or not.
I only tested the activate! and deactivate! methods, since current `::IO` mocks are just forwarding to `::File`

File: test/safe_test.rb [hunk 1]
```diff
@@ -128,4 +128,41 @@ def test_FakeFS_method_deactivates_FakeFS_when_block_raises_exception
...
+  def test_FakeFS_activate_method_stubs_IO_class_if_explicity_asked
+    old_file = ::File
+    old_io = ::IO
+
+    ::FakeFS.activate!(io_mocks: true)
+
+    refute_equal old_file, ::File
+    refute_equal old_io, ::IO
+  ensure
+    ::FakeFS.deactivate!
```
Comment (grosser) [inline test/safe_test.rb:153]: this would be a nicer pattern :)
```
::FakeFS.activate!(io_mocks: true) do
  ... stuff ...
end
```

Comment (grosser) [reply test/safe_test.rb:153]: ... yeah that's another rabbithole where `.with` can't distinguish between these 2 states

**Comment by grosser** [comment]
1.7.0

File: test/safe_test.rb (hunk 1 above)
Comment (Annih) [reply test/safe_test.rb:153]: Yep I gave up on that part :D

**Pull request merged**